weaviate/connect/connection.py

```diff
@@ -253,24 +254,30 @@ def __init__(
 
         # create GRPC channel. If weaviate does not support GRPC, fallback to GraphQL is used.
         if has_grpc and connection_params._has_grpc:
-            s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-            try:
-                s.settimeout(1.0)  # we're only pinging the port, 1s is plenty
-                assert connection_params._grpc_address is not None
-                s.connect(connection_params._grpc_address)
-                s.shutdown(2)
-                s.close()
+            if skip_init_checks:
                 grpc_channel = connection_params._grpc_channel()
                 assert grpc_channel is not None
                 self._grpc_stub = weaviate_pb2_grpc.WeaviateStub(grpc_channel)
                 self._grpc_available = True
-            except (
-                ConnectionRefusedError,
-                TimeoutError,
-                socket.timeout,
-            ):  # self._grpc_stub stays None
-                s.close()
-                self._grpc_available = False
+            else:
+                s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+                try:
+                    s.settimeout(1.0)  # we're only pinging the port, 1s is plenty
+                    assert connection_params._grpc_address is not None
+                    s.connect(connection_params._grpc_address)
+                    s.shutdown(2)
+                    s.close()
+                    grpc_channel = connection_params._grpc_channel()
+                    assert grpc_channel is not None
+                    self._grpc_stub = weaviate_pb2_grpc.WeaviateStub(grpc_channel)
+                    self._grpc_available = True
+                except (
+                    ConnectionRefusedError,
+                    TimeoutError,
+                    socket.timeout,
+                ):  # self._grpc_stub stays None
+                    s.close()
+                    self._grpc_available = False
```

Comment:
how about

```
if not skip_init_checks:
   s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
                try:
                    s.settimeout(1.0)  # we're only pinging the port, 1s is plenty
                    assert connection_params._grpc_address is not None
                    s.connect(connection_params._grpc_address)
                    s.shutdown(2)
                    s.close()
           except:
             ....

grpc_channel = connection_params._grpc_channel()
assert grpc_channel is not None
self._grpc_stub = weaviate_pb2_grpc.WeaviateStub(grpc_channel)
self._grpc_available = True
```

Reply:
Actually, it turns out `_grpc_stub` needs to stay `None` if it isn't available so that the `v3` client can fall back to using GraphQL

This highlights that we should decouple the connections used by the `v3` and `v4` clients and I'll leave that to another PR